ledger/sandbox/src/main/scala/com/digitalasset/platform/indexer/JdbcIndexer.scala

```diff
@@ -26,76 +26,78 @@ import com.digitalasset.platform.metrics.timedFuture
 import com.digitalasset.platform.store.dao.{JdbcLedgerDao, LedgerDao}
 import com.digitalasset.platform.store.entries.{LedgerEntry, PackageLedgerEntry, PartyLedgerEntry}
 import com.digitalasset.platform.store.{FlywayMigrations, PersistenceEntry}
+import com.digitalasset.resources.akka.AkkaResourceOwner
 import com.digitalasset.resources.{Resource, ResourceOwner}
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.control.NonFatal
 
 final class JdbcIndexerFactory(
+    participantId: ParticipantId,
     jdbcUrl: String,
+    actorSystem: ActorSystem,
+    readService: ReadService,
     metrics: MetricRegistry,
 )(implicit logCtx: LoggingContext) {
+
+  private val logger = ContextualizedLogger.get(this.getClass)
+
   def validateSchema()(
       implicit executionContext: ExecutionContext
-  ): Future[InitializedJdbcIndexerFactory] =
+  ): Future[ResourceOwner[JdbcIndexer]] =
     new FlywayMigrations(jdbcUrl)
       .validate()
-      .map(_ => new InitializedJdbcIndexerFactory(jdbcUrl, metrics))
+      .map(_ => initialized)
 
   def migrateSchema(allowExistingSchema: Boolean)(
       implicit executionContext: ExecutionContext
-  ): Future[InitializedJdbcIndexerFactory] =
+  ): Future[ResourceOwner[JdbcIndexer]] =
     new FlywayMigrations(jdbcUrl)
       .migrate(allowExistingSchema)
-      .map(_ => new InitializedJdbcIndexerFactory(jdbcUrl, metrics))
-}
-
-class InitializedJdbcIndexerFactory private[indexer] (
-    jdbcUrl: String,
-    metrics: MetricRegistry,
-)(implicit executionContext: ExecutionContext, logCtx: LoggingContext) {
-  private val logger = ContextualizedLogger.get(this.getClass)
-
-  def owner(
-      participantId: ParticipantId,
-      actorSystem: ActorSystem,
-      readService: ReadService,
-      jdbcUrl: String,
-  ): ResourceOwner[JdbcIndexer] = {
-    val materializer: Materializer = Materializer(actorSystem)
-
-    def fetchInitialState(dao: LedgerDao): Future[Option[Offset]] =
-      for {
-        initialConditions <- readService
-          .getLedgerInitialConditions()
-          .runWith(Sink.head)(materializer)
-        existingLedgerId <- dao.lookupLedgerId()
-        providedLedgerId = domain.LedgerId(initialConditions.ledgerId)
-        _ <- initializeLedger(existingLedgerId, providedLedgerId, dao)
-        initialLedgerEnd <- dao.lookupInitialLedgerEnd()
-      } yield initialLedgerEnd
+      .map(_ => initialized)
 
+  private def initialized(implicit executionContext: ExecutionContext): ResourceOwner[JdbcIndexer] =
```

Comment:
Probably worth adding a first set of (empty) parameters to make sure we're calling something that has an effect.

Reply:
Done.